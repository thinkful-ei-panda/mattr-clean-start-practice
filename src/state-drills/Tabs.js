import React, { Component } from 'react'

export default class Tabs extends Component {
    static defaultProps = { tabs: [] };
    
    state = {
        currentTabIndex: 0
      };

      handleButtonClick = (index) => {
        this.setState({ currentTabIndex: index })
      }

    renderButtons() {
        return this.props.tabs.map((tab, index) => (
          <button key={index} onClick={() => this.handleButtonClick(index)}>
            {tab.name}
          </button>
        ))
      }
      renderContent() {
        // const currentTab = this.props.tabs[0]
        const currentTab = this.props.tabs[this.state.currentTabIndex]
        return (
          <div className='content'>
            <p> {currentTab.content} </p>
          </div>
        )
      }

    render() {
            return (
            <div>
                <h1>This is a Event Handler and Conditionals Drill: Tabs</h1>
                {this.props.tabs.length && (
                    <div className='content'>
                    {this.renderButtons()}
                    {!!this.props.tabs.length && this.renderContent()}
                    </div>
                )}
            </div>
            )
    }
}
