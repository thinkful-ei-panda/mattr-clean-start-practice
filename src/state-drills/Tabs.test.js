// import React from 'react';
// import ReactDOM from 'react-dom';
// import Tabs from './Tabs';

// describe(`Tabs Component`, () => {
//   it('renders without errors', () => {
//     const div = document.createElement('div');
//     ReactDOM.render(<Tabs />, div);
//     ReactDOM.unmountComponentAtNode(div);
//   })
// })

import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer'
import Tabs from './Tabs';

describe(`Tabs Component`, () => {
  // array of objects, each with a name and content
  const tabsProp = [
    { name: 'First tab',
      content: 'This is a sentence for Matt Dizzle' },
    { name: 'Second tab',
      content: 'This is a another for sentence for you.' },
    { name: 'Third tab',
      content: 'Because Lorem Ipsum is too much in this situation.lol' },
  ];

  it('renders without errors', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Tabs />, div);
        ReactDOM.unmountComponentAtNode(div);
    })

  it('renders the first tab by default', () => {
    const tree = renderer.create(<Tabs tabs={tabsProp} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})