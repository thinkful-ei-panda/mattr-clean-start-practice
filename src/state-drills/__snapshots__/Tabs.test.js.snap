// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs Component renders the first tab by default 1`] = `
<div>
  <h1>
    This is a Event Handler and Conditionals Drill: Tabs
  </h1>
  <div
    className="content"
  >
    <button
      onClick={[Function]}
    >
      First tab
    </button>
    <button
      onClick={[Function]}
    >
      Second tab
    </button>
    <button
      onClick={[Function]}
    >
      Third tab
    </button>
    <div
      className="content"
    >
      <p>
         
        This is a sentence for Matt Dizzle
         
      </p>
    </div>
  </div>
</div>
`;
